src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1157,10 +1157,12 @@ void Transformations::MainSnippets() {
     const auto is_infer_prc_supported_by_MHA =
         (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2) &&
          one_of(config.inferencePrecision, ov::element::f32, element::dynamic)) ||
-        (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core) &&
+        ((dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core) ||
+          dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2_vnni_2)) &&
          one_of(config.inferencePrecision, ov::element::bf16, ov::element::f32, element::dynamic)) ||
-        (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx_fp16) &&
-         one_of(config.inferencePrecision, ov::element::f16));
+        ((dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx_fp16) ||
+          dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2_vnni_2)) &&
+         one_of(config.inferencePrecision, ov::element::f16, ov::element::f32, element::dynamic));
```

Comment:
It seems we can reuse some `is_<typename>_supported` things here, can't we?

Reply:
Added the corresponding helpers to `brgemm_utils.hpp`. Now we can reuse it in the plugin. They shows which precisions are supported by implemented snippets-`BRGEMM` on the current platform.

Thanks!